sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentServiceRequestContext.java

```diff
@@ -143,6 +145,14 @@ public void setEndToEndOperationLatencyPolicyConfig(CosmosEndToEndOperationLaten
         this.endToEndOperationLatencyPolicyConfig = endToEndOperationLatencyPolicyConfig;
     }
 
+    public void setIsRequestCancelledOnTimeout(AtomicBoolean isRequestCancelledOnTimeout) {
```

Comment:
for the passed parameter, boolean should be good enough

Reply:
No - I have to set the `AtomicBoolean` itself so as to share it across `DocumentServiceRequestContext` instances.